wgengine/router/router_linux.go

```diff
@@ -164,10 +165,20 @@ func newUserspaceRouterAdvanced(logf logger.Logf, tunname string, linkMon *monit
 		ipt4: netfilter4,
 		ipt6: netfilter6,
 		cmd:  cmd,
-	}, nil
+	}
+
+	return r, nil
 }
 
 func (r *linuxRouter) Up() error {
+	if r.unregLinkMon == nil && r.linkMon != nil {
+		r.unregLinkMon = r.linkMon.RegisterRuleDeleteCallback(func(table uint8, priority uint32) {
+			if priority >= 5200 && priority < 5300 {
+				r.logf("somebody (systemd-networkd?) deleted Tailscale's ip rules; restoring")
+				r.addIPRules()
```

Comment:
is `addIPRules` robust when only some of the rules are deleted?

Reply:
Yes, but its delete itself caused the cleaner to run, so broke it in half with a justAddIPRules part.